cql3/statements/create_keyspace_statement.cc

```diff
@@ -139,6 +143,24 @@ future<> cql3::statements::create_keyspace_statement::grant_permissions_to_creat
     });
 }
 
+future<::shared_ptr<messages::result_message>>
+create_keyspace_statement::execute(service::storage_proxy& proxy, service::query_state& state, const query_options& options) {
+    return schema_altering_statement::execute(proxy, state, options).then([this] (::shared_ptr<messages::result_message> msg) {
+        auto multidc = service::get_storage_service().local().get_token_metadata().
+                                get_topology().get_datacenter_endpoints().size() > 1;
+        auto simple = _attrs->get_replication_strategy_class() == "SimpleStrategy";
+
+        if (multidc && simple) {
+            static const auto warning = "Using SimpleStrategy in a multi-datacenter environment is not recommended.";
+
+            msg->add_warning(warning);
+            _logger.warn(warning);
+        }
+
+        return msg;
```

Comment:
you could do `std::move(msg)`, but then remember to make the lambda mutable first.

Reply:
what about RVO?

Reply:
I have no idea if it applies to lambda parameters too, but I'm also fine with leaving the code as is.

Reply:
Ok I just educated myself more and it seems that RVO doesn't apply here, but there's an implicit move.